experimental/dds/tree2/src/feature-libraries/deltaUtils.ts

```diff
@@ -17,20 +20,56 @@ import { Mutable } from "../util";
  * @param fields - The Map of fields to convert. Not mutated.
  * @param func - The functions used to map tree content.
  */
-export function mapFieldMarks<TIn, TOut>(
-	fields: Delta.FieldMarks<TIn>,
+export function mapFieldsChanges<TIn, TOut>(
+	fields: Delta.FieldMap<TIn>,
 	func: (tree: TIn) => TOut,
-): Delta.FieldMarks<TOut> {
-	const out: Map<FieldKey, Delta.MarkList<TOut>> = new Map();
+): Delta.FieldMap<TOut> {
+	const out: Map<FieldKey, Delta.FieldChanges<TOut>> = new Map();
 	for (const [k, v] of fields) {
-		out.set(k, mapMarkList(v, func));
+		out.set(k, mapFieldChanges(v, func));
 	}
 	return out;
 }
 
 /**
- * Converts a `Delta.MarkList` whose tree content is represented with by `TIn` instances
- * into a `Delta.MarkList`whose tree content is represented with by `TOut` instances.
+ * Converts a `Delta.FieldChanges` whose tree content is represented with by `TIn` instances
+ * into a `Delta.FieldChanges`whose tree content is represented with by `TOut` instances.
+ *
+ * This function is useful for converting `Delta`s that represent tree content with cursors
+ * into `Delta`s that represent tree content with a deep-comparable representation of the content.
+ * See {@link assertMarkListEqual}.
+ * @param fieldChanges - The instance to convert. Not mutated.
+ * @param func - The functions used to map tree content.
+ */
+export function mapFieldChanges<TIn, TOut>(
+	fieldChanges: Delta.FieldChanges<TIn>,
+	func: (tree: TIn) => TOut,
+): Delta.FieldChanges<TOut> {
+	const out: Mutable<Delta.FieldChanges<TOut>> = {};
+	if (fieldChanges.attached !== undefined) {
+		out.attached = mapMarkList(fieldChanges.attached, func);
+	}
+	if (fieldChanges.detached !== undefined) {
+		out.detached = fieldChanges.detached.map(({ id, fields }) => ({
+			id,
+			fields: mapFieldsChanges(fields, func),
+		}));
+	}
+	if (fieldChanges.build !== undefined) {
+		out.build = fieldChanges.build.map(({ id, trees }) => ({
+			id,
+			trees: trees.map(func),
+		}));
+	}
+	if (fieldChanges.rename !== undefined) {
+		out.rename = fieldChanges.rename;
+	}
+	return out;
+}
+
+/**
+ * Converts a list of `Delta.Mark`s whose tree content is represented with by `TIn` instances
+ * into a list of `Delta.Mark`s whose tree content is represented with by `TOut` instances.
```

Comment:
The ungrammatical "represented with by" is used twice here.

Reply:
Oh dear, what happended here?!